pelican/server.py

```diff
@@ -67,34 +66,28 @@ def translate_path(self, path):
 
     def do_GET(self):
         # cut off a query string
-        if '?' in self.path:
-            self.path, _ = self.path.split('?', 1)
-
-        found = False
-        # Try to detect file by applying various suffixes and stripping
-        # patterns.
-        for rstrip_pattern in self.RSTRIP_PATTERNS:
-            if found:
-                break
-            for suffix in self.SUFFIXES:
-                if not hasattr(self, 'original_path'):
-                    self.original_path = self.path
-
-                self.path = self.original_path.rstrip(rstrip_pattern) + suffix
-                path = self.translate_path(self.path)
-
-                if os.path.exists(path):
-                    srvmod.SimpleHTTPRequestHandler.do_GET(self)
-                    logging.info("Found `%s`.", self.path)
-                    found = True
-                    break
-
-                logging.info("Tried to find `%s`, but it doesn't exist.", path)
-
-        if not found:
-            # Fallback if there were no matches
+        original_path = self.path.split('?', 1)[0]
+        # try to find file
+        self.path = self.get_path_that_exists(original_path)
+
+        if not self.path:
             logging.warning("Unable to find `%s` or variations.",
-                            self.original_path)
+                            original_path)
+            return
+
+        logging.info("Found `%s`.", self.path)
+        srvmod.SimpleHTTPRequestHandler.do_GET(self)
```

Comment:
I've never noticed this before... What's the point of this compared to `super(...).do_GET()`?

Reply:
PS: I know, you didn't write that. That code just looks weird to me :).

Reply:
When was `super` introduced? Maybe it's from before that?

Reply:
`super` was added way, waaay before this. It came with new style classes, so around python 2.2 or 2.3 (i think?).